packages/table/src/Table.tsx

```diff
@@ -0,0 +1,196 @@
+
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  React.useEffect(() => {
+    const rowCheckedState: {
+      [n in number]: undefined | boolean;
+    } = {};
+
+    rows.forEach((datum, index) => {
+      if (datum.props.disabled) {
+        return;
+      }
+      rowCheckedState[index] = undefined;
+    });
+
+    if (rowCheckedState) {
+      dispatch({
+        type: Types.SetRowCheckedState,
+        payload: rowCheckedState,
+      });
+    }
+  }, [data]);
+
+  React.useEffect(() => {
+    if (state.rowCheckedState) {
+      const boolArray = Object.values(state.rowCheckedState);
+      const checkSame = boolArray.every(val => val === boolArray[0]);
+
+      if (checkSame) {
+        dispatch({
+          type: Types.ToggleHeaderIndeterminate,
+          payload: false,
+        });
+
+        if (boolArray[0]) {
+          dispatch({
+            type: Types.ToggleHeaderCheckedState,
+            payload: true,
+          });
+        } else {
+          dispatch({
+            type: Types.ToggleHeaderCheckedState,
+            payload: false,
+          });
+        }
+      } else {
+        dispatch({
+          type: Types.ToggleHeaderIndeterminate,
+          payload: true,
+        });
+      }
+    }
+  }, [state.data, state.rowCheckedState]);
+
+  let usingHeaderRow = React.useMemo(() => false, [children]);
+  let rows: Array<React.ReactElement>;
+
+  if (typeof children === 'function') {
+    rows = data.map((datum, index) => children({ datum, index }));
+  }
+
+  const sortRows = (columnId: number, key: string) => {
+    dispatch({
+      type: Types.SortTableData,
+      payload: {
+        columnId,
+        key,
+        data,
+      },
+    });
+  };
+
+  const renderHeader = (array: Array<any>): React.ReactNode => {
+    const cols = array.map((child, index) => {
+      if (isComponentType(child, 'HeaderRow')) {
+        usingHeaderRow = true;
+
+        const { children } = child?.props;
+
+        const renderedChildren = Array.isArray(children)
+          ? children
+          : [children];
+
+        return React.cloneElement(child, {
+          children: renderHeader(renderedChildren),
+        });
+      }
+
+      if (isComponentType(child, 'TableHeader')) {
+        const { label, accessor: accessorProp } = child.props;
+
+        let glyph = 'Unsorted';
+
+        const accessor = accessorProp || label?.toLowerCase();
+
+        if (state.sort?.key?.toLowerCase() === accessor) {
+          glyph =
+            state.sort?.direction === 'asc'
+              ? 'SortAscending'
+              : 'SortDescending';
+        }
+
+        return React.cloneElement(child, {
+          // safe until columns need to be reordered
+          key: index,
+          onClick: sortRows,
+          index,
+          glyph,
+        });
+      }
+
+      if (typeof child === 'string') {
+        return (
+          <TableHeader
+            // safe until columns need to be reordered
+            key={index}
```

Comment:
This is where I'd say to do `child.toLowerCase()` as a nice default.

Reply:
But what if the child is a react element? child.toString().toLowercase() ?

Reply:
Sorry, this works here, but not when cloning the element right above.